jetstream/analysis.py

```diff
@@ -242,30 +242,85 @@ def _calculate_statistics(self, metrics_table: str, period: AnalysisPeriod):
 
         self._publish_view(period, table_prefix="statistics")
 
-    def run(self, current_date: datetime, dry_run: bool) -> None:
-        """
-        Run analysis using mozanalysis for a specific experiment.
-        """
-        self.logger.info(
-            "Analysis.run invoked for experiment %s", self.config.experiment.normandy_slug
-        )
-
+    def is_runnable(self, current_date: Optional[datetime] = None) -> bool:
         if self.config.experiment.normandy_slug is None:
             self.logger.info("Skipping %s; no slug", self.config.experiment.normandy_slug)
-            return  # some experiments do not have a normandy slug
+            return False  # some experiments do not have a normandy slug
 
         if not self.config.experiment.proposed_enrollment:
             self.logger.info(
                 "Skipping %s; no enrollment period", self.config.experiment.normandy_slug
             )
-            return
+            return False
 
         if self.config.experiment.start_date is None:
             self.logger.info("Skipping %s; no start_date", self.config.experiment.normandy_slug)
-            return
+            return False
 
-        if self.config.experiment.end_date and self.config.experiment.end_date < current_date:
+        if (
+            current_date
+            and self.config.experiment.end_date
+            and self.config.experiment.end_date < current_date
+        ):
             self.logger.info("Skipping %s; already ended", self.config.experiment.slug)
+            return False
+
+        return True
+
+    def validate(self) -> None:
+        if not self.is_runnable():
+            raise Exception("Cannot validate experiment")
```

Comment:
Just thinking of how this would integrate with https://github.com/mozilla/jetstream/pull/201/, we might want to raise the Exceptions in `is_runnable`. And maybe rename it to `check_runnable` or something that doesn't sound like it returns a boolean. 
But this looks good for this PR.

Reply:
I had the same thought; I think that'll work nicely!